### ISSUE TITLE
sphinx-apidoc 1.6.6 does not list packages when __init__.py is empty

### ISSUE THREAD

**Issue by dhellmann** [sphinx-doc/sphinx#4446]
Subject: generating API docs for packages with empty __init__.py doesn't work because they do not appear in the submodules toctree list.

### Problem
- Under sphinx 1.6.5 if a package had an empty __init__.py it still had a file generated for its documentation and that package still appeared in the toctree list. That meant the manually managed toctree only needed to include the "modules" file and it would recursively include all of the other files.

#### Procedure to reproduce the problem
```
git clone git://git.openstack.org/openstack/python-openstackclient
cd python-openstackclient
virtualenv venv
source venv/bin/activate
pip install doc/requirements.txt
pip install -e .
sphinx-apidoc . -H Modules -o doc/source/contributor/api/ setup.py openstackclient/volume/v3 openstackclient/tests/ openstackclient/tests/*
cat doc/source/contributor/api/openstackclient.rst
```

Notice that there is no toctree entry under the "Subpackages" section.

#### Error logs / results
```
/opt/devel/projects/osc-doc-build/python-openstackclient/doc/source/contributor/api/openstackclient.api.rst:document isn't included in any toctree
```
- https://docs.openstack.org/python-openstackclient/latest/contributor/api/openstackclient.html

#### Expected results

Generated rst files for packages with empty __init__ should not be ignored in the toctree.

### Reproducible project / your project
- git://git.openstack.org/openstack/python-openstackclient

### Environment info
- OS: Ubuntu Linux 16.04
- Python version: 2.7
- Sphinx version: 1.6.6
- see doc/requirements.txt

**Comment by dhellmann** [comment]
It looks like 2d99648e9982325bbd670da11df5f809e3134284 introduced this change.

**Comment by dhellmann** [comment]
#4449 should fix the problem on the stable branch and #4450 should fix it on master.

**Comment by cank** [comment]
Downgrading to 1.6.5 is indeed a workaround to this.

```pip install sphinx==1.6.5```

**Comment by stephenfin** [comment]
I think this can be closed now, @tk0miya? It was resolved in #4449 so I've guessing this will be fixed in 1.6.7 and 1.7

**Comment by dhellmann** [comment]
@stephenfin it wasn't clear to me whether #4450 (on master) is still needed to ensure we don't have a regression. I'm not familiar with the sphinx development workflow.

**Comment by tk0miya** [comment]
I'll close this after merging #4449 to 1.7-release branch.

**Comment by tk0miya** [comment]
Now merged into 1.7-release and master.
Thanks!

**Comment by proland** [comment]
This appears to be broken again in 1.7

**Comment by tk0miya** [comment]
Confirmed with following Dockerfile:
```
FROM tk0miya/sphinx-html

RUN git clone git://git.openstack.org/openstack/python-openstackclient
WORKDIR /docs/python-openstackclient
RUN pip3 install -r doc/requirements.txt
RUN pip3 install -e .
RUN pip3 install Sphinx==1.7.1
RUN sphinx-apidoc -H Modules -o doc/source/contributor/api/ . setup.py openstackclient/volume/v3 openstackclient/tests/ openstackclient/tests/*
RUN cat doc/source/contributor/api/openstackclient.rst
RUN make -C doc html
```

Sorry, I might made a bug on merging...  I'll take a look later.

**Comment by tk0miya** [comment]
I confirmed this is fixed again in HEAD of 1.7:
```
FROM tk0miya/sphinx-html

RUN git clone git://git.openstack.org/openstack/python-openstackclient
WORKDIR /docs/python-openstackclient
RUN pip3 install -r doc/requirements.txt
RUN apt install -y libffi-dev libssl-dev
RUN pip3 install -e .
RUN pip3 install git+https://github.com/sphinx-doc/sphinx@1.7
RUN sphinx-apidoc -H Modules -o doc/source/contributor/api/ . setup.py openstackclient/volume/v3 openstackclient/tests/ openstackclient/tests/*
RUN cat doc/source/contributor/api/openstackclient.rst
RUN make -C doc html
```

Thanks,
